dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/importer/mapper/ProgramStageInstanceMapper.java

```diff
@@ -65,7 +68,41 @@ public ProgramStageInstance map( Event event )
     {
         ProgramStageInstance psi = workContext.getProgramStageInstanceMap().get( event.getUid() );
 
-        return psi == null ? mapForInsert( event ) : mapForUpdate( event, psi );
+        if ( psi == null )
+        {
+            psi = mapForInsert( event );
+        }
+        else
+        {
+            mapForUpdate( event, psi );
+        }
+
+        // We run the psi through this method to make sure data value
```

Comment:
Perhaps move the Javadoc to the `mapDataValueIdentifier` method, to avoid large inline Javadoc inside a method.